src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/electric/geo/OilPump.java

```diff
@@ -97,24 +97,29 @@ protected void tick(Block b) {
 			}
 		}
 		else {
-			OreGenResource oil = OreGenSystem.getResource("Oil");
-			Chunk chunk = b.getChunk();
-			int supplies = OreGenSystem.getSupplies(oil, chunk, false);
-			
-			if (supplies > 0) {
-				for (int slot : getInputSlots()) {
-					if (SlimefunManager.isItemSimilar(inv.getItemInSlot(slot), new ItemStack(Material.BUCKET), true)) {
+			for (int slot : getInputSlots()) {
+				if (SlimefunManager.isItemSimilar(inv.getItemInSlot(slot), new ItemStack(Material.BUCKET), true)) {
+					OreGenResource oil = OreGenSystem.getResource("Oil");
+					Chunk chunk = b.getChunk();
```

Comment:
Doing this multiple times does not seem like a good idea.

Reply:
Which?

Reply:
getChunk()

Reply:
All of this. You don't need to get the resource, get the chunk and get the supplies every input slot.

Reply:
There should be a break; at the end, then it won't be a problem since it will never run more than once anyway.